### PR TITLE
TMEDIA 858 header block

### PR DESCRIPTION
**Pull request by JackHowa**
## Description

Update headers to use design system v2

## Jira Ticket

- [TMEDIA-858](https://arcpublishing.atlassian.net/browse/TMEDIA-858)

## Acceptance Criteria

Header block is migrated to the new architecture. 

Should be updated to use <HeadingSection> and <Heading> components to render headline, and should use size dropdown to apply the style variations not change heading level

Example of Block changes [arc-themes-blocks/default.jsx at 8bb717fb2b9211154b04007eafbb50ca6c1fe907 · WPMedia/arc-themes-blocks](https://github.com/WPMedia/arc-themes-blocks/blob/8bb717fb2b9211154b04007eafbb50ca6c1fe907/blocks/header-block/features/header/default.jsx) 

Example of CSS changes - [arc-themes-blocks/header.scss at 8bb717fb2b9211154b04007eafbb50ca6c1fe907 · WPMedia/arc-themes-blocks](https://github.com/WPMedia/arc-themes-blocks/blob/8bb717fb2b9211154b04007eafbb50ca6c1fe907/blocks/header-block/features/header/header.scss)

## Test Steps

1. Checkout this branch `git checkout TMEDIA-858-header-block-refactor`
2. Run `npm run storybook`
3. Compare with option 1 and hierarchy for typography in figma 
Screen Shot 2022-05-16 at 13 21 29

### Before

- Hard-coded heading hierarchy 
- Styles determined by news theme css (not block styles)

### After

- Uses helper for establishing hierarchy in the headings from arc themes components
- Uses v2 design system responsive styles based on design discussion

## Dependencies or Side Effects

- manifest https://github.com/WPMedia/arc-themes-manifests/pull/53
- feature pack https://github.com/WPMedia/arc-themes-feature-pack/pull/243

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

# Questions 

-> using heading level from discussion for each level 
Screen Shot 2022-05-16 at 13 21 29

- There's no global font-size for 30px for large size. In the original mocks, they're listed as 30px as well. What should the "large" size be? 32px seems to be what's recommended for enhanced styling docs for reference for default. But the reference there is starting on a different scale and would result in significantly bigger font-size for desktop
Screen Shot 2022-05-13 at 15 13 07

Screen Shot 2022-05-13 at 15 17 57

### CODE REVIEW

File: blocks/header-block/package.json [hunk 1]
```diff
@@ -23,9 +24,7 @@
...
   },
   "peerDependencies": {
-    "@wpmedia/engine-theme-sdk": "*",
-    "@wpmedia/news-theme-css": "*",
-    "@wpmedia/shared-styles": "*"
+    "@wpmedia/arc-themes-components": "*"
   },
   "gitHead": "83fb8d6685958d85ddbfcfbf01d9a5c864cd6c95"
-}
+}
```
Comment (nschubach) [inline blocks/header-block/package.json]:
```suggestion
}

```

File: blocks/header-block/features/header/default.test.jsx [hunk 2]
```diff
@@ -47,81 +36,94 @@ describe("The header block", () => {
...
 		});
 	});
 });
+
+it("if no custom fields provided, should fallback to an empty object and render defaults", () => {
+	const wrapper = mount(<Header />);
+	expect(wrapper.text()).toEqual("");
+	const mediumHeader = wrapper.find(".b-header--medium");
+	expect(mediumHeader).toExist();
+});
```
Comment (nschubach) [inline blocks/header-block/features/header/default.test.jsx]:
```suggestion
describe("when no custom fields provided", () => {
	it("should fallback to an empty object and render defaults", () => {
		const wrapper = mount(<Header />);
		expect(wrapper.text()).toEqual("");
		const mediumHeader = wrapper.find(".b-header--medium");
		expect(mediumHeader).toExist();
	});
});
```

**Review by nschubach — DISMISSED** [review]
Just a few couple changes I see.

File: blocks/header-block/package.json (hunk 1 above)
Comment (nschubach) [reply blocks/header-block/package.json]: This is just adding a newline at the EOF

File: blocks/header-block/features/header/default.test.jsx (hunk 2 above)
Comment (nschubach) [reply blocks/header-block/features/header/default.test.jsx]: Should have the "it" in a "describe"

File: .storybook/themes/news.scss [hunk 3]
```diff
@@ -334,3 +353,4 @@
 
 @use "../../blocks/right-rail-block";
 @use "../../blocks/right-rail-advanced-block";
+@use "../../blocks/header-block";
```
Comment (nschubach) [inline .storybook/themes/news.scss:690]: Can we alphabetize this?  The list is probably going to get long and finding things would be easier.

File: blocks/header-block/features/header/default.test.jsx (hunk 2 above)
Comment (JackHowa) [reply blocks/header-block/features/header/default.test.jsx]: updated

File: .storybook/themes/news.scss (hunk 3 above)
Comment (JackHowa) [reply .storybook/themes/news.scss:690]: updated

**Review by nschubach — APPROVED** [review]

**Pull request merged**